url/url.cpp

```diff
@@ -1460,14 +1464,16 @@ std::optional<std::tuple<std::uint64_t, bool>> UrlParser::parse_ipv4_number(std:
         }
     }
 
+    // TODO(zero-one): Differ width based on largest integer value supported by platform?
     std::uint64_t out;
 
     auto res = std::from_chars(input.data(), input.data() + input.size(), out, r);
 
+    // This deviation from the spec is necessary, because the spec assumes arbitrary precision
```

Comment:
Arbitrary precision sounds like it should only apply to result_out_of_range? Can this be split so it returns failure for invalid argument?